internal/generators/README.md

```
@@ -55,4 +55,23 @@ To add a new generator, follow these steps:
+## Reserved Keywords
+
+Each generator reserves certain symbol names that it exports in the generated output. If a flag key transforms to a reserved name, that flag will be **excluded** from the generated output and a warning will be printed.
+
+| Generator | Reserved names | Transform applied |
+|-----------|---------------|-------------------|
+| Go        | `Client`      | `ToPascal`        |
+| Node.js   | `client`      | `ToCamel`         |
+
+For example, a flag key `"client"` in a Go manifest would transform to `Client` (via `ToPascal`), colliding with the exported `var Client` that the Go generator places in every generated file. The flag will be skipped and the following warning emitted:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)